torchmetrics/utilities/device_dtype_mixin.py

```diff
@@ -0,0 +1,186 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Union
+
+import torch
+from torch.nn import Module
+
+
```

Comment:
what do you even need this for? So far I don't see a usecase here.

Reply:
I am still investigating, I thought it could be used to keep track for the current device for this bug: https://github.com/PyTorchLightning/pytorch-lightning/issues/8227. But couldn't reproduce it yet.

Reply:
It's currently unused in this branch, right? I don't see any uses